export_tsvs.py

```diff
@@ -1,64 +1,138 @@
-ht = ht.annotate(samples=ht.samples_w_het_var.extend(ht.samples_w_hom_var.extend(ht.samples_w_hemi_var)))
-ht = ht.drop(ht.samples_w_het_var, ht.samples_w_hom_var, ht.samples_w_hemi_var)
-
-# explode
-ht = ht.explode(ht.samples).key_by()
-ht = ht.transmute(
-    chrom=ht.locus.contig.replace("chr", ""),
-    pos=ht.locus.position,
-    ref=ht.alleles[0],
-    alt=ht.alleles[1],
-    het_or_hom_or_hemi=ht.samples.het_or_hom_or_hemi,
-    GQ=ht.samples.GQ,
-    S=ht.samples.S,
-)
-
-ht = ht.checkpoint("gs://gnomad/readviz/genomes_v3/gnomad_v3_readviz_crams_exploded.ht", _read_if_exists=True)
-
-print("Output schema:")
-ht.describe()
-
-# write a .tsv for each sample
-for s in sorted(sample_ids):
-    per_sample_ht = ht.filter(ht.S==s, keep=True)
-    per_sample_ht = per_sample_ht.drop(per_sample_ht.S)   # the sample id will be in the .tsv filename, so this column is redundant
-    tsv_output_path = os.path.join(args.output_bucket_path, f"{s}.tsv.bgz") # .replace(' ', '__').replace(":", "_"))
-    print(f"Exporting {tsv_output_path}")
-    per_sample_ht.export(tsv_output_path, header=True) # , parallel="separate_header")
-    
+logging.basicConfig(level=logging.INFO, format='%(asctime)s: %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p')
+
+
+def parse_args():
+	"""Parse command line args."""
+
+	p = argparse.ArgumentParser()
+	p.add_argument(
+		"-o", "--output-bucket-path",
+		help="Path where the .tsvs for each sample will be written",
+		default="gs://gnomad-bw2/gnomad_readviz_tsvs",
+	)
+	p.add_argument(
+		"-p", "--output-partitions",
+		help="Split data for each sample into this many tsv's. Set to 1 to output a single .tsv.bgz file. "
+			 "Setting this >> 1 will produce faster runtimes on large clusters.",
+		default=1000,
+	)
+	p.add_argument(
+		"-f", "--overwrite-checkpoints",
+		action="store_true",
+		help="Overwrite/update all hail checkpoints.",
+	)
+	p.add_argument(
+		"sample_ids_path",
+		help="A text file containing one sample id per line",
+	)
+	args = p.parse_args()
+
+	return args
+
+
+def read_sample_ids(sample_ids_path, n_sample_ids_to_print = 10):
+	"""Read sample ids file.
+
+	Args:
+		sample_ids_path (str): sample ids path
+		n_sample_ids_to_print (int): log no more than this many sample ids to stdout.
+	Return:
+		list: sample id strings
+	"""
+	sample_ids = []
+	with hl.hadoop_open(sample_ids_path) if sample_ids_path.startswith("gs://" ) else open(sample_ids_path, "rt") as f:
+		for i, line in enumerate(f):
+			sample_id = line.rstrip("\n")
+			sample_ids.append(sample_id)
+
+			if i <= n_sample_ids_to_print:
+				logging.info(sample_id)
+				if i == n_sample_ids_to_print and n_sample_ids_to_print > 0:
+					logging.info("...")
+
+	logging.info(f"Parsed {len(sample_ids)} sample ids from {sample_ids_path}")
+
+	return sample_ids
+
+
+def explode_table_by_sample(ht):
+	"""Explode variant-level ht by sample."""
+
+	logging.info("Input schema:")
+	ht.describe()
+
+	ht = ht.annotate(
```

Comment:
This is a very minor change to the structure in the input HT -- can we not just store this information in the desired format in the readviz prep script?

Reply:
Mike did originally, but @lfrancioli suggested it was more space-efficient to store 3 arrays since it's not storing the extra het_or_hom_or_hemi field (which at the time was a string).